interfaces/builtin/network_setup_control.go

```diff
@@ -45,6 +45,18 @@ const networkSetupControlConnectedPlugAppArmor = `
 
 /run/udev/rules.d/ rw,                 # needed for cloud-init
 /run/udev/rules.d/[0-9]*-netplan-* rw,
+
+capability net_admin,
+
+#include <abstractions/dbus-session-strict>
+
+# Allow use of NetPlan Apply API, used to apply network configuration
+dbus (send)
+    bus=system
+    interface=io.netplan.Netplan
+	path=/io/netplan/Netplan
+	member=Apply,
+
```

Comment:
Do you need to provide `peer=unconfined` here? I recall it matters when the method activates the other side.

Reply:
I thought that would be needed as well, but strangely enough the D-Bus rule doesn't work with `peer=(label=unconfined)`. If I had to guess I think it's because the "other side" in this case isn't the unconfined daemon running netplan apply but rather the system bus. It's very possible it's something else however.

Reply:
CC @jdstrand -- correct me if I'm wrong, do we need to duplicate this rule for peer=unconfined or is this good as-is?

Reply:
This interface is currently about exposing netplan apply from the host system to the snap in some fashion. netplan is part of classic/core and runs unconfined (since it isn't supplied as a snap). The types of rules we use in this sistuation are: `dbus (send) ... peer=(label=unconfined),` or sometimes `dbus (send) ... peer=(name=somthing, label=unconfined),`. I would prefer that we use `peer=(label=unconfined)` and understand why that isn't working. Adjusting it and trying and looking at the logged policy violations is the first step.

Reply:
So I debugged this a bit more and discovered that actually for some reason including `peer=(label=unconfined),` is preventing the _initial_ activation, and not subsequent D-Bus calls. If I have not yet activated the netplan apply command over D-Bus, then I get `Access Denied` from busctl when running from inside a snap:

```
$ sudo test-snapd-netplan-apply.netplan apply
Access denied
```

and I see these denials:

```
Aug 14 16:37:38 ian-disco-desktop kernel: audit: type=1400 audit(1565818658.939:125): apparmor="DENIED" operation="capable" profile="snap.test-snapd-netplan-apply.netplan" pid=6739 comm="busctl" capability=12  capname="net_admin"
Aug 14 16:37:38 ian-disco-desktop kernel: audit: type=1107 audit(1565818658.939:126): pid=1572 uid=103 auid=4294967295 ses=4294967295 msg='apparmor="DENIED" operation="dbus_method_call"  bus="system" path="/io/netplan/Netplan" interface="io.netplan.Netplan" member="Apply" mask="send" name="io.netplan.Netplan" pid=6739 label="snap.test-snapd-netplan-apply.netplan"
                                           exe="/usr/bin/dbus-daemon" sauid=103 hostname=? addr=? terminal=?'
```

(note that I removed sys_admin as a capability here, not sure if that's needed for initial activation, it doesn't seem to make a difference)

However, if I activate it outside of confinement, using the exact same busctl command netplan uses internally, it then works from inside confinement:

```
$ sudo busctl call --system io.netplan.Netplan /io/netplan/Netplan io.netplan.Netplan Apply
b true
$ sudo test-snapd-netplan-apply.netplan apply
$ echo $?
0
```

Reply:
Ah, right. netplan needs to adjust its service file in /usr/share/dbus-1/system-services to have: `AssumedAppArmorLabel=unconfined` (see https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/1749000 for related bug).

Reply:
Okay, adding that to the service activate the first time. I will push up here a commit adding the unconfined peer label.